pkg/es/config/config.go

```diff
@@ -188,10 +202,65 @@ func (c *Configuration) GetTagDotReplacement() string {
 }
 
 // GetConfigs wraps the configs to feed to the ElasticSearch client init
-func (c *Configuration) GetConfigs() []elastic.ClientOptionFunc {
+func (c *Configuration) GetConfigs(logger *zap.Logger) []elastic.ClientOptionFunc {
 	options := make([]elastic.ClientOptionFunc, 3)
+	if c.TLS.Enabled {
+		tlsConfig, err := c.CreateTLSConfig(logger)
+		if err != nil {
+			return nil
+		}
+		httpClient := &http.Client{
+			Timeout: c.Timeout,
+			Transport: &http.Transport{
+				TLSClientConfig: tlsConfig,
+			},
+		}
+		options[0] = elastic.SetHttpClient(httpClient)
+		options[1] = elastic.SetURL(c.Servers...)
+		options[2] = elastic.SetSniff(c.Sniffer)
+		return options
+	}
 	options[0] = elastic.SetURL(c.Servers...)
 	options[1] = elastic.SetBasicAuth(c.Username, c.Password)
 	options[2] = elastic.SetSniff(c.Sniffer)
 	return options
 }
+
+// CreateTLSConfig creates TLS Configuration to connect with ES Cluster.
+func (c *Configuration) CreateTLSConfig(logger *zap.Logger) (*tls.Config, error) {
```

Comment:
I think all these new functions can be private. Also, can they be on the TLS config object instead of the top-level Configuration type?

Reply:
Yes. Agree with your point. I will make changes accordingly.